### ISSUE TITLE
Mask R-CNN not learning on pet dataset even without any changes to the code

### ISSUE THREAD

**Issue by matchumen** [tensorflow/tensorflow#30741]
I am attempting to retrain pretrained model on COCO to detect images from Pet Dataset. I followed tutorials and managed to start the training. But according to results, the model is not trained properly.

Steps to recreate:

 1. `git clone https://github.com/cocodataset/cocoapi.git`
 2. `conda create --name tf_test python=3.6`
 3. `conda activate tf_test`
 4. `pip install tensorflow`
 5. `pip install pillow, lxml, jupyter, matplotlib, opencv, contextlib2`
 6. download protoc-3.9.0-win32.zip extract to Program Files\Google Protobuf\
 7. add Program Files\Google Protobuf\bin to path
 8. `for /f %i in ('dir /b object_detection\protos\*.proto') do protoc object_detection\protos\%i --python_out=.`
 9. created environment variable PYTHONPATH with 3 entries: TF\models, TF\models\research\slim, TF\models\research\object_detection
 10. `conda install -c anaconda cython`
 11. `pip install git+https://github.com/philferriere/cocoapi.git#subdirectory=PythonAPI`
 12. restart
 13. `cd TF\models\research\object_detection`
 14. `jupyter notebook`
 15. open object_detection_tutorial.ipynb
 16. added imports:
	from object_detection.utils import label_map_util
	from object_detection.utils import visualization_utils as vis_util
 17. went through demo in object_detection without any problems (I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2)

Directory structure:

 - TF
     + models
     + training 
         * pet_dataset
         * pet_tf_records
         * model
         * checkpoints

Then:

 18. downloaded pet dataset https://www.robots.ox.ac.uk/~vgg/data/pets/
 19. paste contents into pet_dataset
 20. `cd TF`
 21. `conda activate tf_test`
 22. manually set faces_only to False in create_pet_tf_record.py
 23. 
```
python models\research\object_detection\dataset_tools\create_pet_tf_record.py
	--data_dir=training\pet_dataset 
	--output_dir=training\pet_tf_records 
	--mask_type png 
	--label_map_path=models\research\object_detection\data\pet_label_map.pbtxt
```

 24. downloaded mask_rcnn_inception_v2_coco from github to TF\training\model
 25. moved config from models/research/object_detection/sample/configs/ to TF\training\model
 26. changed class number to 37
 27. changed paths accordingly in config file

 28. 
```
python models\research\object_detection\model_main.py 
	--model_dir=training\checkpoints 
	--num_train_steps 1000 
	--pipeline_config_path=training\model\mask_rcnn_inception_v2_coco.config
```
 29. let train for 20 hours

I got following results, when examining in tensorboard, mAP gradually drops from the beginning of the training:
I got following results:

```
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.003
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.019
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.030
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.030
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.030
I0716 07:35:44.988143  3956 evaluation.py:275] Finished evaluation at 2019-07-16-07:35:44
I0716 07:35:44.988143  3956 estimator.py:2039] Saving dict for global step 196: DetectionBoxes_Precision/mAP = 0.00063327036, DetectionBoxes_Precision/mAP (large) = 0.00067928224, DetectionBoxes_Precision/mAP (medium) = -1.0, DetectionBoxes_Precision/mAP (small) = -1.0, DetectionBoxes_Precision/mAP@.50IOU = 0.0025016854, DetectionBoxes_Precision/mAP@.75IOU = 0.0002254508, DetectionBoxes_Recall/AR@1 = 0.018629802, DetectionBoxes_Recall/AR@10 = 0.030065296, DetectionBoxes_Recall/AR@100 = 0.030065296, DetectionBoxes_Recall/AR@100 (large) = 0.030065296, DetectionBoxes_Recall/AR@100 (medium) = -1.0, DetectionBoxes_Recall/AR@100 (small) = -1.0, Loss/BoxClassifierLoss/classification_loss = 0.11927358, Loss/BoxClassifierLoss/localization_loss = 0.10073859, Loss/BoxClassifierLoss/mask_loss = 2.1088743, Loss/RPNLoss/localization_loss = 0.11095625, Loss/RPNLoss/objectness_loss = 0.54912615, Loss/total_loss = 2.988967, global_step = 196, learning_rate = 0.0002, loss = 2.988967
```

Please, do you have any suggestions where did i go wrong?

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow-tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
